Debuggin rails application 4 (#24)
campuscode/rails-guides-pt-BR: pt-BR/debugging_rails_applications.md

**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**

Debugando Aplicações Rails
============================

Esse guia introduz técnicas de debugação para aplicações de Ruby on Rails

Após ler esse guia, você saberá:

> * O proposito de debugar

Reviewer: ```suggestion
* O propósito de debugar
```

**NÃO LEIA ESTE ARQUIVO NO GITHUB, OS GUIAS SÃO PUBLICADOS NO https://guiarails.com.br.**

Esse guia introduz técnicas de debugação para aplicações de Ruby on Rails

Após ler esse guia, você saberá:

* O propósito de debugar
* Como encontrar problemas nas suas aplicações que testes não estão identificando
* As diferentes maneiras de debugar
* Como analisar o *stack trace*

> =======

Reviewer 2: ```suggestion
```

Acho que no merge essa linha ficou perdida

Você pode especificar um logger alternativo em `config/application.rb` ou em qualquer outro arquivo de ambiente, por exemplo: 

```ruby
config.logger = Logger.new(STDOUT)
config.logger = Log4r::Logger.new("Application Log")
```

Ou na sessão `Initializer`, adicione _qualquer_ um dos seguintes: 

```ruby
Rails.logger = Logger.new(STDOUT)
Rails.logger = Log4r::Logger.new("Application Log")
```
DICA: Por padrão, cada log é criado em `Rails.root/log/` e o arquivo de log é criado com o nome do ambiente no qual a aplicação está sendo executada.

### Níveis de Log

Quando algo é logado, o log é feito no arquivo de log correspondente se o nível de log da mensagem for igual ou maior do que o nível de log configurado. Se você quiser saber o nível de log atual, você pode chamar o método `Rails.logger.level`.

Os níveis de log disponiveis são: `:debug`, `:info`, `:warn`, `:error`, `:fatal`,
e `:unknown`, correspondendo aos níveis de log de 0 até 5, respectivamente. Para mudar o nível de log padrão, utilize:

```ruby
config.log_level = :warn # Em qualquer inicializador de ambiente, ou
Rails.logger.level = 0 # a qualquer momento
```

> Isso é útil quando você quer criar logs em ambientes diferentes de desenvolvimento, para não sobrecarregar os logs do seu aplicativo com informação desnecessária.

Reviewer 2: ```suggestion
Isso é útil quando você quer criar logs em ambientes diferentes de desenvolvimento ou preparação sem sobrecarregar os logs do seu aplicativo com informação desnecessária.
```

Rails.logger = Logger.new(STDOUT)
Rails.logger = Log4r::Logger.new("Application Log")
```

> DICA: Por padrão, cada log é criado em `Rails.root/log/` e o arquivo de log é criado com o nome do ambiente no qual a aplicação está sendo executada.

Reviewer 2: ```suggestion

DICA: Por padrão, cada log é criado em `Rails.root/log/` e o arquivo de log é criado com o nome do ambiente no qual a aplicação está sendo executada.
```

Reviewer 2: Coloquei a linha de cima de volta, o que acha?

### Níveis de Log 

> Quando algo é logado, o log é feito no arquivo de log correspondente se o nível de log da mensagem for igual ou maior do que o nível de log configurado. Se você quiser saber o nível de log atual, você pode chamar o método `Rails.logger.level`.

Reviewer 2: ```suggestion
Quando algo é adicionado ao log é feito no arquivo de log correspondente se o nível de log da mensagem for igual ou maior do que o nível de log configurado. Se você quiser saber o nível de log atual, você pode chamar o método `Rails.logger.level`.
```

Reviewer 3: ```suggestion
Quando algo é registrado, a informação é armazenada no local de registro correspondente se o nível de *log* for igual ou maior que o configurado. Se você quiser saber o nível atual do registro, você pode o método `Rails.logger.level`.
```

Eu entendo que *log* seja uma palavra técnica bastante específica, mas sugiro usar a palavra registro, pelo menos em algumas ocorrências, assim a gente evita repetição e usa um termo que existe já em português que faz sentido dentro do contexto.

Os níveis de log disponiveis são: `:debug`, `:info`, `:warn`, `:error`, `:fatal`,
e `:unknown`, correspondendo aos níveis de log de 0 até 5, respectivamente. Para mudar o nível de log padrão, utilize:

```ruby
config.log_level = :warn # Em qualquer inicializador de ambiente, ou
Rails.logger.level = 0 # a qualquer momento
```

> Isso é útil quando você quer criar logs em ambientes diferentes de desenvolvimento ou preparação sem sobrecarregar os logs do seu aplicativo com informação desnecessária.

Reviewer 2: ```suggestion
Isso é útil quando você quer criar logs em ambientes diferentes de desenvolvimento ou homologação sem sobrecarregar os logs do seu aplicativo com informação desnecessária.
```

Author: @HenriqueMorato , eu acho que o @akaninja deu uma ideia boa pra essa parte. Registro encaixa bem e evita essa repetição. Deixei a sugestão dele nesse ponto e vou ver se acho outros lugares que da pra trocar também.

Reviewer 2: Claro! Pode dar resolve aqui. Sem problemas :)

Author: @HenriqueMorato mudei pra homologação nesse ultimo commit que dei e te coloquei de co autor.
